Removed consistency check from redo log processing.

dutow opened the pull request:
This caused an error when the redo log references an unknown tablespace,
breaking partial backups.

With this change, we keep the diagnostic message about this in the logs,
but do not fail the prepare operation.

The testcases also had to be adjusted, as MySQL 8.0 stores the table list
in the mysql database, which we restore in its original point. The tables
missing from the backup are still visible, but can't be used now.

gl-sergei commented:
Shall we remove missing tablespaces from MySQL data dictionary and from InnoDB system tables?

dutow commented:
I'm sure that in some cases, we'll have to modify the system tables, but this looks like a different case to me.

In this test case, we backup the mysql database, and some of our tables, and then restore this backup.

So if we'd like to remove the missing entries, we have two choices:
* when doing a partial backup, and if the mysql db is part of this backup, remove the unnecessary entries. With this, we would backup a modified version of the mysql db instead of the original - for example, what if a user only backs up the mysql db? Should we remove everything else? I don't think this would be a good idea, at least not always.
* when doing the copy-back, as we can assume that the target data dir is empty - but is it safe to assume that the user won't restore those files in some other way? This looks like a safer approach, but I'm still not sure about it.

gl-sergei commented:
There was no MySQL data dictionary in 5.7, so things need to change for MySQL 8. But here is how it worked.

1. There were SYS_* tables keeping InnoDB data dictionary
2. There were .frm files

All traces of missing tables were cleaned up by xtrabackup:

1. During prepare xtrabackup removed entries for missing tables from SYS_* tables.
2. During the backup, *.frm files for missing tables simply were not copied.

As a result, after copy-back, MySQL did not see any phantom tables in SHOW TABLES, SHOW CREATE TABLE, etc. I.e. if someone wanted specified only single backup in `databases/tables/include` pattern, after restore he would see just that table.

-----

With this patch tables remain visible in SHOW TABLES output and SELECT from missing table throws an error, which is not how it used to work with 5.x. What will happen if we will try to remove missing table from MySQL CLI?

I think we must mandatory backup all necessary system tables. If there is a table in `mysql` database that contains data dictionary, we must back it up. Then, we will be able to delete missing entries from this table during prepare.

dutow committed:
Removed consistency check from redo log processing

Also updated related test to match the now inconsistent data dictionary.
As we don't update the table list, the tables remain visible, but can't be used for selects.
Dropping these tables also works.

dutow commented:
Restored the original version without the DD changes.
